obspy/io/nordic/core.py

```diff
@@ -0,0 +1,1379 @@
+        new_event.magnitudes.append(Magnitude())
+        new_event.magnitudes[2].mag = _float_conv(topline[72:75])
+        new_event.magnitudes[2].magnitude_type = topline[75]
+        new_event.magnitudes[2].creation_info = \
+            CreationInfo(agency_id=topline[76:79].strip())
+        new_event.magnitudes[2].origin_id = new_event.origins[0].\
+            resource_id
+    # convert the nordic notation of magnitude to more general notation
+    for _magnitude in new_event.magnitudes:
+        _magnitude.magnitude_type = _nortoevmag(_magnitude.magnitude_type)
+    # Set the useful things like preferred magnitude and preferred origin
+    new_event.preferred_origin_id = str(new_event.origins[0].resource_id)
+    if len(new_event.magnitudes) > 1:
+        try:
+            # Select moment first, then local, then
+            mag_filter = ['MW', 'Mw', 'ML', 'Ml', 'MB', 'Mb',
+                          'MS', 'Ms', 'Mc', 'MC']
+            _magnitudes = [(m.magnitude_type, m.resource_id)
+                           for m in new_event.magnitudes]
+            preferred_magnitude = sorted(_magnitudes,
+                                         key=lambda x: mag_filter.index(x[0]))
+            new_event.preferred_magnitude_id = str(preferred_magnitude[0][1])
+        except ValueError:
+            # If there is a magnitude not specified in filter
+            new_event.preferred_magnitude_id =\
+                str(new_event.magnitudes[0].resource_id)
+    elif len(new_event.magnitudes) == 1:
+        new_event.preferred_magnitude_id =\
+            str(new_event.magnitudes[0].resource_id)
+    return new_event
+
+
+def read_spectral_info(sfile):
+    """
+    Read spectral info from an sfile.
+
+    :type sfile: str
+    :param sfile: Sfile to read from.
+
+    :returns: dictionary of spectral information, units as in seisan manual, \
+        expect for logs which have been converted to floats.
+    """
+    with open(sfile, 'r') as f:
+        spec_inf = _read_spectral_info(f=f)
+    return spec_inf
+
+
+def _read_spectral_info(f):
+    """Internal spectral reader.
+
+    :type f: file
+    :param f: File open in read mode.
+
+    :returns: dictionary of spectral information, units as in seisan manual, \
+        expect for logs which have been converted to floats.
+    """
+    event = _readheader(f=f)
+    f.seek(0)
+    origin_date = UTCDateTime(event.origins[0].time.date)
+    relevant_lines = []
+    for line in f:
+        if str(line[1:5]) == str('SPEC'):
+            relevant_lines.append(line)
+    spec_inf = []
+    if len(relevant_lines) == 0:
+        return spec_inf
+    for line in relevant_lines:
+        spec_str = str(line.rstrip().lstrip())
+        if str(spec_str[5:12]) == str('AVERAGE'):
+            info = {}
+            info['station'] = str('AVERAGE')
+            info['channel'] = str('')
+            info['moment'] = _float_conv(spec_str[16:22])
+            if info['moment'] == 999:
```

Comment:
Anywhere there's some condition, the `_float_conv` makes things more complicated than a simple `try:/except ValueError:`.